tools/Azure.Mcp.Tools.Communication/src/Services/CommunicationService.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Communication.Sms;
+using Azure.Mcp.Core.Options;
+using Azure.Mcp.Core.Services.Azure;
+using Azure.Mcp.Tools.Communication.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.Communication.Services;
+
+public class CommunicationService(ILogger<CommunicationService> logger) : ICommunicationService
+{
+    private readonly ILogger<CommunicationService> _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+    public async Task<List<SmsResult>> SendSmsAsync(
+        string connectionString,
+        string from,
+        string[] to,
+        string message,
+        bool enableDeliveryReport = false,
+        string? tag = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        ValidateRequiredParameters(connectionString, from, to, message);
+
+        try
+        {
+            var smsClient = new SmsClient(connectionString);
+
+            var sendOptions = new SmsSendOptions(enableDeliveryReport)
+            {
+                Tag = tag
+            };
+
+            _logger.LogInformation("Sending SMS from {From} to {ToCount} recipient(s)", from, to.Length);
+
+            var response = await smsClient.SendAsync(
+                from: from,
+                to: to,
+                message: message,
+                options: sendOptions);
+
+            var results = new List<SmsResult>();
+            foreach (var result in response.Value)
+            {
+                results.Add(new SmsResult
+                {
+                    MessageId = result.MessageId,
+                    To = result.To,
+                    Successful = result.Successful,
+                    HttpStatusCode = result.HttpStatusCode,
+                    ErrorMessage = result.ErrorMessage
+                });
+
+                _logger.LogInformation("SMS to {To}: Success={Success}, MessageId={MessageId}, Status={Status}",
+                    result.To, result.Successful, result.MessageId, result.HttpStatusCode);
+            }
+
+            return results;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Failed to send SMS from {From} to {ToCount} recipient(s)", from, to.Length);
+            throw;
+        }
+    }
+
+    private static void ValidateRequiredParameters(string connectionString, string from, string[] to, string message)
```

Comment:
You shouldn't need this as we have similar method

Reply:
done